examples/python/ml/jax_kmeans.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2022 Ant Group Co., Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Start nodes.
+# > bazel run -c opt //examples/python/utils:nodectl -- up
+#
+# Run this example script.
+# > bazel run -c opt //examples/python/ml:jax_kmeans
+
+
+import argparse
+import json
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+
+import spu.binding.util.distributed as ppd
+
+parser = argparse.ArgumentParser(description="distributed driver.")
+parser.add_argument("-c", "--config", default="examples/python/conf/3pc.json")
+args = parser.parse_args()
+
+with open(args.config, "r") as file:
+    conf = json.load(file)
+
+ppd.init(conf["nodes"], conf["devices"])
+
+
+def kmeans(data_alice, data_bob, k, random_numbers, n_epochs=10):
+    # data: data set
+    # k: number of clusters
+    # return: cluster centers, labels
+
+    data = jnp.append(data_alice, data_bob, axis=0)
+
+    n = data.shape[0]
+    m = data.shape[1]
+
+    # initialize cluster centers
+    centers = jnp.zeros((k, m))
+
+    def loop_for_1(i, stat):
+        centers = stat
+        centers = centers.at[i].set(data[random_numbers[i]])
+        return centers
+
+    centers = jax.lax.fori_loop(0, k, loop_for_1, centers)
+
+    # initialize labels
+    labels = jnp.zeros(n)
+
+    def loop_while_func(stat):
+
+        rounds, centers, new_centers, labels = stat
```

Comment:
original `centers` seems to be never used, why should we pass it into `loop_while_func`?

Reply:
> original `centers` seems to be never used, why should we pass it into `loop_while_func`?

Oops, this is because in the original version without jax, the condition that exit the while loop depends on the norm of two "centers", like:

```
if np.linalg.norm(centers - new_centers) < 1e-6:
    break
```

This feature cannot be used in jax, so I make the loop exits in constant rounds. But I forget to remove them. I'll fix it. Thank you.

Reply:
If there's any more problem, please tell me any time. May I commit them together if there's no big bugs?